colossalai/auto_parallel/tensor_shard/solver/solver.py

```diff
@@ -160,6 +160,7 @@ def _prepare_data_for_solver(self):
                     compute_cost = compute_cost_item.total
                     memory_cost = memory_cost_item.total
 
+                memory_cost = memory_cost.parameter + memory_cost.activation + memory_cost.buffer
```

Comment:
In line 157 and line 161, the variable `memory_cost` has been assigned, why do we overwrite it here?

Reply:
I will add some comments to explain it. In short, the memory_cost in 157 and 161 is in type of MemoryCost, and what solver wants is int, so I extract the related values in line 163.

Reply:
Ok, I see. It is kind of dizzy to read without the comments haha